network/dag/bbolt_dag.go

```diff
@@ -227,42 +306,53 @@ func (dag *bboltDAG) Add(ctx context.Context, transactions ...Transaction) error
 	return nil
 }
 
-func (dag bboltDAG) Walk(ctx context.Context, algo WalkerAlgorithm, visitor Visitor, startAt hash.SHA256Hash) error {
+func (dag bboltDAG) Walk(ctx context.Context, visitor Visitor, startAt hash.SHA256Hash) error {
 	return bboltTXView(ctx, dag.db, func(ctx context.Context, tx *bbolt.Tx) error {
 		transactions := tx.Bucket([]byte(transactionsBucket))
-		nexts := tx.Bucket([]byte(nextsBucket))
-		if transactions == nil || nexts == nil {
+		clocksBucket := tx.Bucket([]byte(clockBucket))
+		if transactions == nil {
 			// DAG is empty
 			return nil
 		}
-		return algo.walk(ctx, visitor, startAt, func(hash hash.SHA256Hash) (Transaction, error) {
```

Comment:
I believe the `BFSAlgorithmWalker` stopped at TXs that missed payload (correct me if I'm wrong), is that still the case?

Reply:
Yes, the pushback of the resume point is now at the 2 different callback funcs (payload and tx)